### PR TITLE
Only accept known locations in request.errors.add() (fixes #99)

### LINKED ISSUE #99: Check that the errors contain the right parameters
For instance, we should cry if we put "headers" instead of "header" when doing `request.errors.add()`.

### PR DESCRIPTION
**Pull request by leplatrem**
If this conflicts with your PR @tarekziade, do not take it into account, I'll rebase it later.
r?

### CODE REVIEW

**Comment by leplatrem** [comment]
From what I see in #376, it should probably also support `cookie`, `path`, `method`

**Comment by leplatrem** [comment]
@Natim r?

File: cornice/tests/test_errors.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from cornice.errors import Errors
+
+
+class TestErrorsHelper(TestCase):
+    def setUp(self):
+        self.errors = Errors()
+
+    def test_raises_an_exception_when_location_is_unsupported(self):
+        with self.assertRaises(ValueError):
+            self.errors.add('something')
```
Comment (Natim) [inline cornice/tests/test_errors.py]: Can we add tests for each accepted parameters?

File: cornice/errors.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def add(self, location, name=None, description=None, **kw):
         """Registers a new error."""
         allowed = ('body', 'querystring', 'url', 'header', 'path',
                    'cookies', 'method')
-        if location not in allowed:
+        if location != '' and location not in allowed:
```
Comment (Natim) [inline cornice/errors.py]: Do you really want to accept empty location?

File: cornice/errors.py [hunk 3]
```diff
@@ -16,6 +16,11 @@ def __init__(self, status=400):
 
     def add(self, location, name=None, description=None, **kw):
         """Registers a new error."""
+        allowed = ('body', 'querystring', 'url', 'header', 'path',
+                   'cookies', 'method')
+        if location != '' and location not in allowed:
```
Comment (Natim) [inline cornice/errors.py]: Do you really want to accept empty location?

Comment (leplatrem) [inline cornice/errors.py]: That's what colander gives when you raise a validation error that is not related to a specific node (eg. `provide param either via header or querystring`)
